spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/partitioner/CassandraPartitionGenerator.scala

```diff
@@ -79,12 +79,18 @@ private[connector] class CassandraPartitionGenerator[V, T <: Token[V]](
     val tokenRanges : Seq[TokenRange] = if (partitionMap != null) {
       // Compute map from (start) token to corresponding hosts based on the partition map.
       // Converting token map into Sequence of TokenRanges describing the ring for this table.
-      partitionMap.values.toSeq.map { partition =>
+      var ranges = partitionMap.values.toSeq.map { partition =>
         val startToken = tokenFactory.tokenFromString(PartitionAwarePolicy.YBToCqlHashCode(partition.getStartKey()).toString)
         val endToken = tokenFactory.tokenFromString(PartitionAwarePolicy.YBToCqlHashCode(partition.getEndKey()).toString)
         val hosts = partition.getHosts().map(_.getAddress).toSet
         new TokenRange(startToken, endToken, hosts, tokenFactory)
       }
+      ranges match {
+        case head #:: rest =>
+          if (head.replicas.isEmpty)
+            ranges = describeRing
```

Comment:
why do we only need to check the head here, what about the other entries?

Reply:
The issue had no replica in partition map. In this case checking the head would be enough - we know whether there is  replica returned or not.